src/test/java/seedu/address/model/person/UniquePersonListTest.java

```diff
@@ -47,6 +50,31 @@ public void contains_personWithSameIdentityFieldsInList_returnsTrue() {
         assertTrue(uniquePersonList.contains(editedAlice));
     }
 
+    @Test
+    public void containsSameIdPerson_nullPerson_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniquePersonList.containsSameIdPerson(null));
+    }
+
+    @Test
+    public void containsSameIdPerson_personNotInList_returnsFalse() {
+        assertFalse(uniquePersonList.containsSameIdPerson(ALICE));
+    }
+
+    @Test
+    public void containsSameIdPerson_differentId_returnsFalse() {
+        uniquePersonList.add(ALICE);
+        Person editedPerson = new PersonBuilder(ALICE).withId(VALID_ID_BOB).build();
+        assertFalse(uniquePersonList.containsSameIdPerson(editedPerson));
+    }
+
+    @Test
+    public void containsSameIdPerson_sameIdDifferentIdentity_returnsTrue() {
+        uniquePersonList.add(ALICE);
+        Person editedPerson = new PersonBuilder(ALICE).withName(VALID_NAME_BOB).withPhone(VALID_PHONE_BOB)
+                .build();
+        assertTrue(uniquePersonList.containsSameIdPerson(editedPerson));
+    }
+
```

Comment:
Some additional tests can be conducted in this class. Such as tests for adding unidentifiable locations and tests for setting location lists which dont fulfill the identifiable property. You may refer to #118 UniqueLocationListTest for reference.

Reply:
More tests added.

Reply:
One more test was missed.

Reply:
Test added.